scripts/build_test.py

```diff
@@ -14,29 +14,270 @@
 # limitations under the License.
 
+import StringIO
+import collections
+import os
+import random
+import re
+import subprocess
+import threading
+
 # pylint: disable=relative-import
 import build
 from core.tests import test_utils
 
 # pylint: enable=relative-import
 
+# Override Pylint's protected access rule due to multiple private functions in
+# the file.
+# pylint: disable=protected-access
+
 
 class BuildTests(test_utils.GenericTestBase):
     """Test the build methods."""
 
+    def test_minify(self):
+        """Tests _minify with an invalid filepath."""
+        with self.assertRaises(subprocess.CalledProcessError) as calledProcess:
+            build._minify(
+                'invalid/path/to/input.js', 'invalid/path/to/output.js')
+        calledProcessException = calledProcess.exception
+        # returncode is the exit status of the child process.
+        self.assertEqual(calledProcessException.returncode, 1)
+
+    def test_join_files(self):
+        """Determine third_party.js contains the content of the first 10 JS
+        files in /third_party/static.
+        """
+        # Prepare a file_stream object from StringIO.
+        third_party_js_stream = StringIO.StringIO()
+        # Get all filepaths from manifest.json.
+        dependency_filepaths = build.get_dependencies_filepaths()
+        # Join and write all JS files in /third_party/static to file_stream.
+        build._join_files(dependency_filepaths['js'], third_party_js_stream)
+        counter = 0
+        # Only checking first 10 files.
+        JS_FILE_COUNT = 10
+        for js_filepath in dependency_filepaths['js']:
+            if counter == JS_FILE_COUNT:
+                break
+            with open(js_filepath, 'r') as js_file:
+                # Assert that each line is copied over to file_stream object.
+                for line in js_file:
+                    self.assertIn(line, third_party_js_stream.getvalue())
+            counter += 1
+
+    def test_minify_and_create_sourcemap(self):
+        """Tests _minify_and_create_sourcemap with an invalid filepath."""
+        with self.assertRaises(subprocess.CalledProcessError) as calledProcess:
+            build._minify_and_create_sourcemap(
+                'invalid/path/to/input.js', 'invalid/path/to/output.js')
+        calledProcessException = calledProcess.exception
+        # returncode is the exit status of the child process.
+        self.assertEqual(calledProcessException.returncode, 1)
+
+    def test_generate_copy_tasks_for_fonts(self):
+        """Test _generate_copy_tasks_for_fonts to ensure that the number of copy
+        tasks matches the number of font files.
+        """
+        copy_tasks = collections.deque()
+        # Get all filepaths from manifest.json.
+        dependency_filepaths = build.get_dependencies_filepaths()
+        # Setup a sandbox folder for copying fonts.
+        target_fonts_dir = os.path.join('target', 'fonts', '')
```

Comment:
Maybe let's make this more obviously sandbox-y: target_fonts_dir = 'test_target'

And then, at the end, assert that each task contains that string, perhaps?

Reply:
Renamed!